src/main/java/com/criticalsoftware/AnnouncementResource.java

```diff
@@ -0,0 +1,118 @@
+package com.criticalsoftware;
+
+import jakarta.inject.Inject;
+import jakarta.ws.rs.*;
+import jakarta.ws.rs.core.Response;
+import org.bson.types.ObjectId;
+
+@Path("/announcements")
+public class AnnouncementResource {
+
+    @Inject
+    private AnnouncementRepository repository;
+
+    @Inject
+    private UserRepository userRepository;
+
+    @DELETE
+    @Path("/{id}")
+    public Response delete(@PathParam("id") String id) {
+
+        // Check if the provided ID is valid
+        if (!id.matches("[a-fA-F0-9]{24}")) {
```

Comment:
As you use this string several times in this class, you can replace it with a constant.